### PR TITLE
[FLINK-31447][runtime] Aligning unit tests of FineGrainedSlotManager with DeclarativeSlotManager

### PR DESCRIPTION
**Pull request by huwh**
## What is the purpose of the change
Aligning unit tests of FineGrainedSlotManager with DeclarativeSlotManager
The unit test diff between FGSM and DSM are list in : https://docs.google.com/spreadsheets/d/1km5gwECvdC0pu6HwvjBfBdHB4RIXYn5ls7AggHkH3Hk/edit#gid=0

## Brief change log
  - *Let ResourceManagerTest run with DeclarativeSlotManager and FineGrainedSlotManager*
  - *Add some unit tests for FineGrainedSlotManager*

## Verifying this change
  - *This change is only about unit tests*

## Does this pull request potentially affect one of the following parts:
  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)

### CODE REVIEW

**Review by reswqa — COMMENTED** [review]
Thanks @huwh, I left some comments, please take a look.

**Comment by reswqa** [comment]
`RuntimeManagerTest` should be changed to `ResourceManagerTest` in the first commit message.

**Comment by huwh** [comment]
Thanks @reswqa , I have updated this pr.

File: flink-runtime/src/test/java/org/apache/flink/runtime/resourcemanager/slotmanager/DefaultSlotStatusSyncerTest.java [hunk 1]
```diff
@@ -114,6 +114,64 @@ void testAllocateSlot() throws Exception {
...
+                        "address",
+                        ResourceProfile.ANY);
+        final AllocationID allocationId = requestFuture.get().f2;
+        assertThat(resourceTracker.getAcquiredResources(jobId))
+                .contains(ResourceRequirement.create(ResourceProfile.ANY, 1));
+        assertThat(taskManagerTracker.getAllocatedOrPendingSlot(allocationId))
+                .hasValueSatisfying(slot -> assertThat(slot.getJobId()).isEqualTo(jobId));
+        assertThat(taskManagerTracker.getAllocatedOrPendingSlot(allocationId))
+                .hasValueSatisfying(
+                        slot -> assertThat(slot.getState()).isEqualTo(SlotState.PENDING));
```
Comment (reswqa) [inline flink-runtime/src/test/java/org/apache/flink/runtime/resourcemanager/slotmanager/DefaultSlotStatusSyncerTest.java:166]: These two assertions should be able to be combined into one `hasValueSatisfying `.

**Review by reswqa — APPROVED** [review]
Thanks @huwh for the update, LGTM. Let's squash all fix-up commit.

**Pull request closed without merging**